File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/AttributeSetInstanceExtensionFieldIdFlattenedDto.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\FlattenedDtoInterface;

class AttributeSetInstanceExtensionFieldIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'groupId' => 'string',
            'index' => 'string',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var AttributeSetInstanceExtensionFieldId
     */
    private $value;

    /**
     * @param AttributeSetInstanceExtensionFieldId $value
     */
    public function __construct(AttributeSetInstanceExtensionFieldId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new AttributeSetInstanceExtensionFieldId();
        }
    }

    /**
     * @return AttributeSetInstanceExtensionFieldId
     */
    public function toAttributeSetInstanceExtensionFieldId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getGroupId()
    {
        return $this->value->getGroupId();
    }

    /**
     * @param string $groupId
     */
    public function setGroupId($groupId)
    {
        $this->value->setGroupId($groupId);
    }

    /**
     * @return string
     */
    public function getIndex()
    {
        return $this->value->getIndex();
    }

    /**
     * @param string $index
     */
    public function setIndex($index)
    {
        $this->value->setIndex($index);
    }

    public function toString()
    {
        $pValues = [
            $this->getGroupId(),
            $this->getIndex(),
        ];
        return implode(',', $pValues);
    }

    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setGroupId($pValues[0]);
        $this->setIndex($pValues[1]);
    }

}

File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/AttributeSetInstanceExtensionFieldIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class AttributeSetInstanceExtensionFieldIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "groupId",
            "index",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private AttributeSetInstanceExtensionFieldId value;

    public AttributeSetInstanceExtensionFieldIdFlattenedDto()
    {
        this(new AttributeSetInstanceExtensionFieldId());
    }

    public AttributeSetInstanceExtensionFieldIdFlattenedDto(AttributeSetInstanceExtensionFieldId value)
    {
        this.value = value;
    }

    public AttributeSetInstanceExtensionFieldId toAttributeSetInstanceExtensionFieldId()
    {
        return this.value;
    }


    public String getGroupId()
    {
        return this.value.getGroupId();
    }

    public void setGroupId(String groupId)
    {
        this.value.setGroupId(groupId);
    }

    public String getIndex()
    {
        return this.value.getIndex();
    }

    public void setIndex(String index)
    {
        this.value.setIndex(index);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        AttributeSetInstanceExtensionFieldIdFlattenedDto other = (AttributeSetInstanceExtensionFieldIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

